src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -105,21 +109,36 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     var resourceList = new HashSet<string>();
                     foreach (var param in notYetIndexedParams)
                     {
-                        if (param.TargetResourceTypes != null)
+                        foreach (var baseResourceType in param.BaseResourceTypes)
                         {
-                            resourceList.UnionWith(param.TargetResourceTypes);
-                        }
+                            if (baseResourceType == "Resource")
+                            {
+                                resourceList.UnionWith(_modelInfoProvider.GetResourceTypeNames().ToHashSet());
 
-                        if (param.BaseResourceTypes != null)
-                        {
-                            resourceList.UnionWith(param.BaseResourceTypes);
+                                // We added all possible resource types, so no need to continue
+                                break;
```

Comment:
>break [](start = 32, length = 5)

It feels brittle to me. Who controls order in `BaseResourceTypes`? We can stop populating `resourceList` if "Resource" show up as first type.

Reply:
If the first value in `param.BaseResourceTypes` is "Resource", we add every type to `resourceList`, since a search parameter with this base type will affect all resources.

Not breaking out of this for-loop wouldn't cause any harm, but it would be inefficient since we would be attempting to add the same thing more than once.

Reply:
I've re-read code, and I agree with you.
There is still corner case like {"Bundle", "DomainResource"} vs {"DomainResource","Bundle"} I think it would produce different resourceList, but my guess no one will do that.

---
In reply to: [597055066](https://github.com/microsoft/fhir-server/pull/1753#discussion_r597055066) [](ancestors = 597055066)

Reply:
Good catch - even if it's unlikely I think it makes sense to consider it. Fixed in acab74a.